samples/e2eTestUtils/ConfigUtils.ts

```diff
@@ -0,0 +1,55 @@
+import fs from 'fs';
+
+export class StringReplacer {
+    private filePath: string;
+    private mappings: Record<string, string>;
+
+    constructor(filePath: string) {
+        this.filePath = filePath;
+        this.mappings = {};
+    }
+
+    /**
+     * Updates the file by replacing all instances of keys with values in the map
+     * @param map: key-value pairs of strings to replace
+     */
+    async replace(map: Record<string, string>) {
```

Comment:
This all feels a bit overkill to me. Did you consider putting the config in a JSON file instead? That would make it much simpler to update values in place or even just replace the whole file.

Reply:
What I had in mind was a solution that satisfies these: 

1) Doesn't require us hardcode things: so IDs don't get leaked, people don't try to sign-in to these apps, if things change on the lab side we don't have to manually update configs and etc.
2) Doesn't sacrifice presentation: while samples here are primarily for testing, they are still the first introduction for a lot of people. Putting stuff in js files like authConfig.js allows us to place comments, import types, constants, enums and etc.
3) Doesn't require change to the existing code: loading configs from json means we'll have to async initialize msal, and sometimes have to configure framework settings to play nice with json imports.

I'm happy to reconsider though. We're already doing this in node samples and a few js ones, so there's precedent. I just think this is more flexible going forward.

Reply:
I think I prefer the static file approach. It feels cleaner and less brittle to me. Something else to keep in mind is that we use these samples locally to test our changes. Hiding the details behind the lab API will make it more cumbersome for us to get up and running. 

1. To address the leaking of details, they're not secret, I think as long as they're not the default and we clearly document they are for internal use only I don't personally have a problem with them being visible. To address the maintenance concern we could perhaps take a hybrid approach where the lab API is used in the E2E environment but the static config is still available for us to use locally. But how often do we think the lab API changes? Is it worth the extra effort/cost?
2. To address json vs js, I don't have a strong opinion. If you want to define the config in js files it just means we replace (move/rename) the whole file with the alternate config rather than replacing individual values. That's fine too.
3. Sounds like using the js files from point 2 would address these concerns as well

Reply:
I would like the "edit-ability" that a static file provides. If the `lab` goes down, a static file can be replaced easily with any config for our local testing and makes debug easy too. That is my only issue with this approach.